### ISSUE TITLE
Run system reads in parallel when possible

### ISSUE THREAD

**Issue by kolanos** [iopipe/iopipe-python#113]
The system reads can be run in the background via `multiprocessing.Queue` and collected at the end of the invocation. This way the filesystem reads are non-blocking for the handler itself. This also serves as an optimization for system reads in general.

**Comment by kolanos** [comment]
Use `concurrent.futures` for running in parallel:
https://docs.python.org/3/library/concurrent.futures.html

**Comment by kolanos** [comment]
Closing in favor of #209.
